porth.py

```diff
@@ -955,14 +956,23 @@ def usage(compiler_name: str):
                 basename = basename[:-len(porth_ext)]
             basedir = path.dirname(program_path)
         basepath = path.join(basedir, basename)
-
+        print(f"{basepath=} {basedir=} {basename=}")
         print("[INFO] Generating %s" % (basepath + ".asm"))
         program = compile_file_to_program(program_path, include_paths);
         generate_nasm_linux_x86_64(program, basepath + ".asm")
-        cmd_call_echoed(["nasm", "-felf64", basepath + ".asm"])
-        cmd_call_echoed(["ld", "-o", basepath, basepath + ".o"])
+        if platform.system() == "Windows": # Pseudo Windows 10/11 support, requiring a wsl(version irelevant) installation with nasm and ld (gnu binutils) install on the default Distro
+            win_basepath = basepath.replace("\\", "/")
+            cmd_call_echoed(["wsl", "nasm", "-felf64", win_basepath + ".asm"])
+            cmd_call_echoed(["wsl", "ld", "-o", win_basepath, win_basepath + ".o"])
```

Comment:
Do you need to prefix the commands with `wsl` if you are running everything from within the WSL console?

Reply:
The trick I am doing, is that you don't have to run it from wsl, since wsl is a completely contained linux distro, It was able to run porth before this change fine, but for ease of devolopment and tool uage on windows (configuriung wsl for IDE's or toolchains it pretty annoying) It now works if you run it from anywhere, without havin to worry to specifically run it through wsl.

Reply:
Now it autodectects if not run through wsl (since wsl will return "Linux" for platform.system()) and will then switch to run all the external commands through wsl, while still running on the windows host itself.

Reply:
Do I understand correctly that if you simply run everything from within WSL no changes in the code are required at all?

Reply:
Yes, you understand me correctly.

Reply:
But as I said, configuring tools and other systems to run through wsl is often very hard, annoying or straight up impossible.